src/components/Messages/CreateMessageModal/SearchBarMessages.tsx

```diff
@@ -0,0 +1,155 @@
+import SearchRoundedIcon from "@mui/icons-material/SearchRounded";
+import {
+  Autocomplete,
+  Avatar,
+  Box,
+  IconButton,
+  InputAdornment,
+  List,
+  ListItemAvatar,
+  ListItemButton,
+  ListItemText,
+  TextField,
+  Typography,
+} from "@mui/material";
+import { OtherUser } from "./MessagesModalList";
+import { useAppSelector } from "../../../state/hooks";
+import { useEffect, useState } from "react";
+import axios from "axios";
+import { useNavigate } from "react-router-dom";
+const styles = {
+  box: {
+    margin: "auto",
+    alignItems: "center",
+    justifyContent: "center",
+    padding: 1,
+    width: "95%",
+  },
+  searchField: {
+    "& .MuiOutlinedInput-root": {
+      borderRadius: "50px",
+    },
+  },
+  avatar: { margin: "auto" },
+  displayName: {
+    fontWeight: "bold",
+  },
+  primaryTextContainer: {
+    gap: 0.5,
+    width: "30%",
+    height: "100%",
+  },
+  stack: {
+    display: "flex",
+    flexDirection: "row",
+    width: "100%",
+  },
+  autocomplete: { "&.MuiAutocomplete-input": { paddingLeft: 0 } },
+  listItemButton: {
+    "&:hover": {
+      backgroundColor: "transparent",
+    },
+  },
+  searchIcon: { paddingRight: 0 },
+};
+
+type SearchBarProps = {
+  placeholder: string;
+  setFocusSearchBar: (state: boolean) => void;
+  onClose: () => void;
+};
+
+const SearchBarMessages = ({
+  placeholder,
+  setFocusSearchBar,
+  onClose,
+}: SearchBarProps) => {
+  const user = useAppSelector((state) => state.user);
+  const [followedList, setFollowedList] = useState<OtherUser[]>([]);
+
+  const navigate = useNavigate();
+  const routeChange = (otherUser: OtherUser) => {
+    onClose();
+    const path = `/messages/${user.userId}/${otherUser.otherUserId}`;
+    navigate(path);
+  };
+
+  useEffect(() => {
+    const fetchDMList = async () => {
+      const result = await axios.get(
+        "http://localhost:3001/api/messages/followedList",
+        {
+          params: {
+            userId: user.userId,
+          },
+        }
+      );
+      setFollowedList(result.data as OtherUser[]);
+    };
+    fetchDMList();
+  }, [user]);
+
+  return (
+    <Box sx={styles.box}>
+      <Autocomplete
+        onOpen={() => setFocusSearchBar(true)}
+        onClose={() => setFocusSearchBar(false)}
```

Comment:
Instead of passing `setFocusSearchBar` as a prop, can we include this state change in the `onClose` function and `onOpen` function and have those as props instead? That way all `onClose` functionality is wrapped up in one function

Reply:
So you want me to send `onOpen` and `onClose` as props to `MessagesSearchBar.tsx`? 

Would look like 
```suggestion
        onOpen={onOpen}
        onClose={onClose}
```

Reply:
Resolved this offline!